core/embed/rust/src/ui/model_tt/component/fido.rs

```diff
@@ -0,0 +1,210 @@
+use core::ops::Deref;
+
+use crate::ui::{
+    component::{Child, Component, Event, EventCtx, Image, Label},
+    display,
+    geometry::{Alignment, Insets, Rect},
+    model_tt::component::{
+        fido_icons::get_fido_icon_data,
+        swipe::{Swipe, SwipeDirection},
+        theme, ScrollBar,
+    },
+};
+
+use super::CancelConfirmMsg;
+
+const ICON_HEIGHT: i16 = 70;
+const SCROLLBAR_INSET_TOP: i16 = 5;
+const SCROLLBAR_HEIGHT: i16 = 10;
+const APP_NAME_PADDING: i16 = 12;
+const APP_NAME_HEIGHT: i16 = 30;
+
+pub enum FidoMsg {
+    Confirmed(usize),
+    Cancelled,
+}
+
+pub struct FidoConfirm<F: Fn(usize) -> T, T, U> {
+    page_swipe: Swipe,
+    app_name: Label<T>,
+    account_name: Label<T>,
+    icon: Child<Image>,
+    /// Function/closure that will return appropriate page on demand.
+    get_account: F,
+    scrollbar: ScrollBar,
+    fade: bool,
+    controls: U,
+}
+
+impl<F, T, U> FidoConfirm<F, T, U>
+where
+    F: Fn(usize) -> T,
+    T: Deref<Target = str> + From<&'static str>,
```

Comment:
It's just because `Layout` is built with `Deref`, so either i would need to change `Layout` to `AsRef` (which propagates to a lot of places), or, easier, switch Fido to `Deref`.

More generally, it seems that `Deref` is the more appropriate of the two for us. IIUC, "Deref" means "this thing can be treated as a `&str`" whereas "AsRef" means "this thing is capable of giving you a `&str` if you ask for it". My ultimate goal is to replace `T: Deref` with a global type alias `StrType` which is going to be `StrBuffer` if we build with micropython, or `&str` if not, and for that case the `Deref`-based code seems to require less changes.